nomad/datasource_nomad_job.go

```diff
@@ -0,0 +1,287 @@
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceJob() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceJobRead,
+		Schema: map[string]*schema.Schema{
+
+			"id": {
+				Description: "Job ID",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			// computed attributes
+			"name": {
+				Description: "Job Name",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"namepsace": {
+				Description: "Job Namespace",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"type": {
+				Description: "Job Type",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"version": {
+				Description: "Job Version",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"region": {
+				Description: "Job Region",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"datacenters": &schema.Schema{
+				Description: "Job Datacenters",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"status": {
+				Description: "Deployment Status",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+			"status_description": {
+				Description: "Deployment Status Description",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+			"submit_time": {
+				Description: "Job Submit Time",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"create_index": {
+				Description: "Create Index",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"modify_index": {
+				Description: "Modify Index",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"job_modify_index": {
+				Description: "Job Modify Index",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"stop": {
+				Description: "Job Stopped",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"priority": {
+				Description: "Job Priority",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"parent_id": {
+				Description: "Job Parent ID",
+				Computed:    true,
+				Type:        schema.TypeString,
+			},
+			"task_groups": {
+				Description: "Job Task Groups",
+				Computed:    true,
+				Type:        schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"placed_canaries": {
+							Type:     schema.TypeString,
+							Optional: true,
```

Comment:
I think these should all be `Computed`, not `Optional`. I don't know what the behavior of an `Optional` field in a `Computed` block is, but the semantics of it don't really make sense. `Optional` indicates you can supply a value but aren't required to. `Computed` without `Optional` or `Required` prohibits supplying a value.